drivers/storage/portworx/deployment.go

```diff
@@ -0,0 +1,670 @@
+			hostPath:  "/var/run/dbus",
+			mountPath: "/var/run/dbus",
+		},
+	}
+
+	// dockerVolumeInfoList is a list of volumes applicable when running in docker
+	dockerVolumeInfoList = []volumeInfo{
+		{
+			name:      "dev",
+			hostPath:  "/dev",
+			mountPath: "/dev",
+		},
+		{
+			name:      "optpwx",
+			hostPath:  "/opt/pwx/bin",
+			mountPath: "/export_bin",
+		},
+		{
+			name:      "dockerplugins",
+			hostPath:  "/run/docker/plugins",
+			mountPath: "/run/docker/plugins",
+		},
+		{
+			name:             "libosd",
+			hostPath:         "/var/lib/osd",
+			mountPath:        "/var/lib/osd",
+			mountPropagation: mountPropagationPtr(v1.MountPropagationBidirectional),
+		},
+		{
+			name:      "hostproc",
+			hostPath:  "/proc",
+			mountPath: "/hostproc",
+		},
+		{
+			name:      "src",
+			hostPath:  "/usr/src",
+			mountPath: "/usr/src",
+			coreos: &coreosVolumeInfo{
+				hostPath:  "/lib/modules",
+				mountPath: "/lib/modules",
+			},
+		},
+		{
+			name:             "kubelet",
+			hostPath:         "/var/lib/kubelet",
+			mountPath:        "/var/lib/kubelet",
+			mountPropagation: mountPropagationPtr(v1.MountPropagationBidirectional),
+			openshift: &openshiftVolumeInfo{
+				hostPath:  "/var/lib/origin/openshift.local.volumes",
+				mountPath: "/var/lib/origin/openshift.local.volumes",
+			},
+		},
+	}
+
+	// csiVolumeInfoList is a list of volumes applicable when running csi sidecars
+	csiVolumeInfoList = []volumeInfo{
+		{
+			name:         "registration-dir",
+			hostPath:     "/var/lib/kubelet/plugins_registry",
+			hostPathType: hostPathTypePtr(v1.HostPathDirectoryOrCreate),
+		},
+		{
+			name:         "csi-driver-path",
+			hostPath:     "/var/lib/kubelet/plugins/com.openstorage.pxd",
+			hostPathType: hostPathTypePtr(v1.HostPathDirectoryOrCreate),
+		},
+	}
+)
+
+type template struct {
+	cluster         *corev1alpha1.StorageCluster
+	isDocker        bool
+	isPxDev         bool
+	runOnMaster     bool
+	isPKS           bool
+	isCoreOS        bool
+	isOpenshift     bool
+	imagePullPolicy v1.PullPolicy
+	startPort       int
+}
+
+func newTemplate(cluster *corev1alpha1.StorageCluster) (*template, error) {
+	if cluster == nil {
+		return nil, fmt.Errorf("storage cluster cannot be empty")
+	}
+
+	t := &template{cluster: cluster}
+
+	enabled, err := strconv.ParseBool(cluster.Annotations[annotationIsPKS])
+	t.isPKS = err == nil && enabled
+
+	enabled, err = strconv.ParseBool(cluster.Annotations[annotationIsCoreOS])
+	t.isCoreOS = err == nil && enabled
+
+	enabled, err = strconv.ParseBool(cluster.Annotations[annotationIsOpenshift])
+	t.isOpenshift = err == nil && enabled
+
+	enabled, err = strconv.ParseBool(cluster.Annotations[annotationIsDocker])
+	t.isDocker = err == nil && enabled
+
+	enabled, err = strconv.ParseBool(cluster.Annotations[annotationIsPxDev])
+	t.isPxDev = err == nil && enabled
+
+	enabled, err = strconv.ParseBool(cluster.Annotations[annotationRunOnMaster])
+	t.runOnMaster = err == nil && enabled
```

Comment:
we should not continue support for runOnMaster. Running on master was a requirement during 1.6 timeframe and we have no longer see any use cases.

Reply:
Removing support for px dev and running on master.